src/main/filters/SpectralTilt.cpp

```diff
@@ -0,0 +1,370 @@
+
+            }
+
+            if (enNorm == STLT_NORM_AUTO)
+            {
+                if (fSlopeNepNep <= 0)
+                    enNorm = STLT_NORM_AT_DC;
+                else
+                    enNorm = STLT_NORM_AT_NYQUIST;
+            }
+
+            if (fLowerFrequency >= 0.5f * nSampleRate)
+                fLowerFrequency = DFL_LOWER_FREQUENCY;
+
+            if (fUpperFrequency >= 0.5f * nSampleRate)
+                fUpperFrequency = DFL_UPPER_FREQUENCY;
+
+            if (fLowerFrequency >= fUpperFrequency)
+            {
+                fLowerFrequency = DFL_LOWER_FREQUENCY;
+                fUpperFrequency = DFL_UPPER_FREQUENCY;
+            }
+
+            if ((enSlopeUnit == STLT_SLOPE_UNIT_NONE) || (fSlopeNepNep == 0.0f))
+            {
+                bBypass = true;
+                bSync = true;
+                return;
+            }
+            else
+            {
+                bBypass = false;
+            }
+
+            float l_angf = 2.0f * M_PI * fLowerFrequency;
+            float u_angf = 2.0f * M_PI * fUpperFrequency;
+
+            // Exponential spacing ratio for poles.
+            float r = powf(u_angf / l_angf, 1.0f / (nOrder - 1));
+            float c_fn = bilinear_coefficient(1.0f, nSampleRate);
+            float negZero = l_angf * powf(r, -fSlopeNepNep);
+            float negPole = l_angf;
+
+            // We have nOrder bilinears. We combine them 2 by 2 to get biquads.
+
+            sFilter.begin();
+            for (size_t n = 0; n < nOrder; ++n)
+            {
+                if (n % 2 != 0)
+                    continue;
+
+                bilinear_spec_t spec_now = compute_bilinear_element(negZero, negPole);
+                negZero *= r;
+                negPole *= r;
+
+                bilinear_spec_t spec_nxt = compute_bilinear_element(negZero, negPole);
+                negZero *= r;
+                negPole *= r;
+
+                dsp::biquad_x1_t *digitalbq = sFilter.add_chain();
+                if (digitalbq == NULL)
+                    return;
+
+                dsp::f_cascade_t analogbq;
+                analogbq.t[0] = spec_now.b0 * spec_nxt.b0;
+                analogbq.t[1] = spec_now.b0 * spec_nxt.b1 + spec_now.b1 * spec_nxt.b0;
+                analogbq.t[2] = spec_now.b1 * spec_nxt.b1;
+                analogbq.b[0] = spec_now.a0 * spec_nxt.a0;
+                analogbq.b[1] = spec_now.a0 * spec_nxt.a1 + spec_now.a1 * spec_nxt.a0;
+                analogbq.b[2] = spec_now.a1 * spec_nxt.a1;
+
+                dsp::bilinear_transform_x1(digitalbq, &analogbq, c_fn, 1);
+
+                // The denominator coefficients in digitalbq will have opposite sign with respect the maths.
+                // This is correct, as this is the LSP convention.
+            }
+            sFilter.end(true);
+
+            bSync = true;
+        }
+
+        void SpectralTilt::process_add(float *dst, const float *src, size_t count)
+        {
+            if (src != NULL)
```

Comment:
Also I think we can refactor the code in the same way as we did for `process_mul()`.

Reply:
Similar refactors can be done for `Velvet` and `NoiseGenerator` classes methods too. I'll go for it.